superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterOption/AdhocFilterOption.test.tsx

```diff
@@ -19,14 +19,35 @@
 import React from 'react';
 import { render, screen, waitFor } from 'spec/helpers/testing-library';
 import userEvent from '@testing-library/user-event';
-import { HTML5Backend } from 'react-dnd-html5-backend';
-import { DndProvider } from 'react-dnd';
 import AdhocFilter, {
   EXPRESSION_TYPES,
   CLAUSES,
 } from 'src/explore/components/controls/FilterControl/AdhocFilter';
 import AdhocFilterOption, { AdhocFilterOptionProps } from '.';
 
+jest.mock('src/components/Icons/Icon', () => ({
+  __esModule: true,
+  default: ({
+    fileName,
+    role,
+    iconColor,
+    ...rest
+  }: {
+    fileName: string;
+    role: string;
+    iconColor: string;
+  }) => (
+    <span
+      role={role ?? 'img'}
+      aria-label={fileName.replace('_', '-')}
+      // @ts-ignore
```

Comment:
This is to prevent the following warning in the test:
```
Warning: React does not recognize the `iconColor` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `iconcolor` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
        in span (created by default)
        ...
```

This seemed arbitrary to the functionality so I felt it was safe to ignore.

Reply:
The `iconColor` or the `rest` props are not used in the tests so it should be safe to remove them when mocking.

You can use this code and the tests will remain green:
```
jest.mock('src/components/Icons/Icon', () => ({
  __esModule: true,
  default: ({ fileName, role }: { fileName: string; role: string }) => (
    <span role={role ?? 'img'} aria-label={fileName.replace('_', '-')} />
  ),
  StyledIcon: () => <span />,
}));
```

Reply:
That's much cleaner, thanks! Fixed in [`this commit`](https://github.com/apache/superset/pull/22915/commits/b4c2371b09bb63ef76021a02faa7d7b1f7968f39).